scripts/workers/submission_worker.py

```diff
@@ -676,9 +676,29 @@ def process_submission_callback(body):
                 SUBMISSION_LOGS_PREFIX, body
             )
         )
+        submission_pk = json.loads(body)["submission_pk"]
         body = yaml.safe_load(body)
         body = dict((k, int(v)) for k, v in body.items())
-        process_submission_message(body)
+        pushgateway_endpoint = os.environ.get("PUSHGATEWAY_ENDPOINT")
+        response = requests.post(
+            "{endpoint}/metrics/job/{j}/instance/{i}/submission_pk/{s}".format(
```

Comment:
It is not a good practice to just put the request in the main block. Please create a separate method for this counter update. And why are we making a explicit post request for deque counting? We should use Prometheus client.

Reply:
Prometheus Pushgateway does not provide a support for using multiprocess registries which have been used in our project due to multiple workers in django... so if I use prometheus client,I would need to specify a registry. If I don't use the multiprocess registry, the results would be inconsistent as each worker would have an instance of registry.. So after some debugging and reading, I switched to API calls which give consistent and accurate results

Reply:
@burnerlee for workers you don't have to worry about multiprocessing. We only have 1 worker per challenge at the moment. We can add support for multiprocessing later